mne/viz/backends/_pysurfer_mayavi.py

```diff
@@ -423,3 +420,39 @@ def _save_figure(img, filename):
     FigureCanvasAgg(fig)
     fig.figimage(img, resize=True)
     fig.savefig(filename)
+
+
+def _close_3d_figure(figure):
+    from mayavi import mlab
+    mlab.close(figure)
+
+
+def _take_3d_screenshot(figure, mode='rgb', filename=None):
+    from mayavi import mlab
+    from mne.viz.backends.renderer import MNE_3D_BACKEND_TESTING
+    if MNE_3D_BACKEND_TESTING:
+        ndim = 3 if mode == 'rgb' else 4
+        if figure.scene is None:
+            figure_size = (600, 600)
+        else:
+            figure_size = figure.scene._renwin.size
+        return np.zeros(tuple(figure_size) + (ndim,), np.uint8)
+    else:
+        from pyface.api import GUI
+        gui = GUI()
+        gui.process_events()
+        with warnings.catch_warnings(record=True):  # traits
+            img = mlab.screenshot(figure, mode=mode)
+            if isinstance(filename, str):
+                _save_figure(img, filename)
+            return img
```

Comment:
Only nest what you have to in the context manager. So probably just the mlab line